packages/instantsearch.js/src/lib/utils/createSendEventForHits.ts

```diff
@@ -31,13 +31,13 @@ const buildPayloads = ({
   widgetType,
   methodName,
   args,
-  isSearchStalled,
+  invalidStatus,
```

Comment:
Right now, this flag seems to be more accurately called `shouldSendViewEvent`.

Why don't we instead pass the InstantSearch instance and keep the logic here?

Reply:
That's possible too, I didn't want to touch the code too much, but happy to change if you think it should be refactored

Reply:
It's debt we don't have to clean up later (aka never) so since we have good code coverage, I think we should use the opportunity.

Reply:
I've made a draft PR here: https://github.com/algolia/instantsearch/pull/5439